bdk_core/Cargo.toml

```diff
@@ -9,12 +9,14 @@ edition = "2021"
 bitcoin = { version = "0.28" }
 miniscript = { git =  "https://github.com/llfourn/rust-miniscript", rev = "2d351c08caca292e8710d74b950bc200f5a539cc" }
 serde_crate = { package = "serde", version = "1", optional = true, features = ["derive"] }
-
+hashbrown = { version = "0.12.1" , optional = true }
 
 [features]
-default = ["std"]
+default = []
```

Comment:
It got me a little confused when `--no-default-features` wasn't working.. And that's happening because by default we are linking both `alloc` and `std` and some things are being linked from `std` which should have been linked from `alloc`. Both `std` and `alloc` have same names for stuffs.. And this is not being caught in static analysis because we have `std` in `default` thus `#no_std` attribute isn't working its magic.. `no_std` gives static error anytime we link to `std` libs even by mistake..

So it made sense to me to keep the lib by default `no-std` if that's the intention as per lib definition?? And remove such compiler confusion in future.. Users can ask for `std` specifically if they want it..

Reply:
From what I can tell the only time we linked to `std` before this PR was with `vec!` which you fixed by also adding the `#[macro_use]` to `alloc`. This is all that was required to fix `--no-default-features`. I think the lib should use things from `std` by default and only be `no_std` if you ask it to be by disabling default features. This is a common pattern FWIW.

Reply:
Oh okay.. In that case it makes sense.. I thought the pattern was to be no_std only and include std when only needed.. So in that case should we also link to `std::Vec`, `std::Box` etc in case std is enabled??

Reply:
`std::Box` and `std::Vec` are the same one from `alloc` so it doesn't matter.